interfaces/builtin/microstack_support_test.go

```diff
@@ -0,0 +1,158 @@
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/interfaces/kmod"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	"github.com/snapcore/snapd/interfaces/udev"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type microStackSupportInterfaceSuite struct {
+	iface    interfaces.Interface
+	slotInfo *snap.SlotInfo
+	slot     *interfaces.ConnectedSlot
+	plugInfo *snap.PlugInfo
+	plug     *interfaces.ConnectedPlug
+}
+
+const microStackSupportMockPlugSnapInfoYaml = `name: microstack
+version: 1.0
+apps:
+ app:
+  command: foo
+  plugs: [microstack-support]
+`
+
+var _ = Suite(&microStackSupportInterfaceSuite{
+	iface: builtin.MustInterface("microstack-support"),
+})
+
+func (s *microStackSupportInterfaceSuite) SetUpTest(c *C) {
+	s.slotInfo = &snap.SlotInfo{
+		Snap:      &snap.Info{SuggestedName: "core", SnapType: snap.TypeOS},
+		Name:      "microstack-support",
+		Interface: "microstack-support",
+	}
+	s.slot = interfaces.NewConnectedSlot(s.slotInfo, nil, nil)
+	plugSnap := snaptest.MockInfo(c, microStackSupportMockPlugSnapInfoYaml, nil)
+	s.plugInfo = plugSnap.Plugs["microstack-support"]
+	s.plug = interfaces.NewConnectedPlug(s.plugInfo, nil, nil)
+}
+
+func (s *microStackSupportInterfaceSuite) TestName(c *C) {
+	c.Assert(s.iface.Name(), Equals, "microstack-support")
+}
+
+func (s *microStackSupportInterfaceSuite) TestUsedSecuritySystems(c *C) {
+	// connected plugs have a non-nil security snippet for apparmor
+	apparmorSpec := &apparmor.Specification{}
+	err := apparmorSpec.AddConnectedPlug(s.iface, s.plug, s.slot)
+	c.Assert(err, IsNil)
+	c.Assert(apparmorSpec.SecurityTags(), HasLen, 1)
+
+	// connected plugs have a non-nil security snippet for seccomp
+	seccompSpec := &seccomp.Specification{}
+	err = seccompSpec.AddConnectedPlug(s.iface, s.plug, s.slot)
+	c.Assert(err, IsNil)
+	c.Assert(seccompSpec.Snippets(), HasLen, 1)
+}
+
+func (s *microStackSupportInterfaceSuite) TestConnectedPlugSnippet(c *C) {
+	apparmorSpec := &apparmor.Specification{}
+	err := apparmorSpec.AddConnectedPlug(s.iface, s.plug, s.slot)
+	c.Assert(err, IsNil)
+	c.Assert(apparmorSpec.SecurityTags(), DeepEquals, []string{"snap.microstack.app"})
+	c.Assert(apparmorSpec.SnippetForTag("snap.microstack.app"), testutil.Contains, "/dev/vhost-net rw,\n")
```

Comment:
Please add a couple of more lines to this check, preferably those which are microstack-specific (`/dev/microstack-*/{,**} rw,` sounds like a good candidate)

Reply:
Done.

Reply:
Done https://github.com/snapcore/snapd/pull/8926/commits/d84dd2f166a3eac250c1e44d14f1b5d31e25f363